### PR TITLE
feat: (Platform) introduce Slider component

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
https://github.com/sap/fundamental-ngx/issues/4443
#### Please provide a brief summary of this pull request.
Introduce Platform Slider component
#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [X] update `README.md`
- [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

**Review by kavya-b — COMMENTED** [review]
please take a look at these minor comments

File: apps/docs/src/app/platform/component-docs/platform-slider/examples/base/slider-basic-example.component.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/core';
+import { SliderChangeEvent } from '@fundamental-ngx/platform';
+
+@Component({
+    selector: 'fdp-slider-basic-example',
```
[the remark this answers is not in the document]
Comment (Betrozov) [reply apps/docs/src/app/platform/component-docs/platform-slider/examples/base/slider-basic-example.component.ts:5]: @kavya-b When we use fdp, doesn't that mean it's a platform? :slightly_smiling_face:

File: apps/docs/src/app/platform/component-docs/platform-slider/examples/base/slider-basic-example.component.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    templateUrl: './slider-basic-example.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
```
[the remark this answers is not in the document]
Comment (Betrozov) [reply apps/docs/src/app/platform/component-docs/platform-slider/examples/base/slider-basic-example.component.ts:7]: it is not needed if it is not used in the component
it works on StackBlitz

File: apps/docs/src/app/platform/component-docs/platform-slider/examples/base/slider-basic-example.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kavya-b) [reply apps/docs/src/app/platform/component-docs/platform-slider/examples/base/slider-basic-example.component.ts:5]: True 😄  But I thought I saw stackblitz service class was filtering on `fdp-platform`; however, I re-checked and it is just `fdp` so you can ignore this comment :)

File: apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+@Component({
+    selector: 'fdp-slider-form-field-example',
+    templateUrl: './slider-form-field-example.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    styles: [
+        `
+            .slider-example .fd-container.fd-form-layout-grid-container .fd-row .fd-col {
+                overflow: initial !important;
```
Comment (kavya-b) [inline apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.ts:12]: `initial` also will not be supported by IE ;) can we not just keep `auto`?

**Review by IlyaSurmay — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.ts (hunk 3 above)
Comment (Betrozov) [reply apps/docs/src/app/platform/component-docs/platform-slider/examples/form-field/slider-form-field-example.component.ts]: It not helped
Will try `visible`

**Review by KevinOkamoto — APPROVED** [review]

**Review by kavya-b — APPROVED** [review]
LGTM!

File: libs/platform/src/lib/components/slider/slider.component.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    Optional,
+    Output,
+    Self,
+    SkipSelf,
+    ViewEncapsulation
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+
+import { SliderControlValue, SliderCustomValue } from '@fundamental-ngx/core';
+export { SliderControlValue, SliderCustomValue } from '@fundamental-ngx/core';
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/slider/slider.component.ts:17]:
```suggestion
```

**Review by dimamarksman — APPROVED** [review]
Please remove a duplication

**Review by dimamarksman — APPROVED** [review]

**Comment by puru-hk** [comment]
Looks good

**Review by puru-hk — APPROVED** [review]

**Pull request merged**